File: 代码随想录训练营/day1.cpp
#include <vector>
#include <iostream>
using namespace std;

int search(vector<int>& nums, int target)
{
    int l = 0;
    int r =nums.size() - 1;
    while( l <= r)
    {
        int mid = ((r -l)>>1) + l;
        if(nums[mid] == target) return mid;
        if(nums[mid] < target) l = mid + 1;
        else r = mid - 1;
    }
    return -1;
}

int main()
{
    
    int n;
    cin >> n;
    vector<int> nums(n);
    for(int i = 0; i < n; i++){
        cin >> nums[i];
    }
    int target = 9;
    cout << search(nums, target) << endl;
}

File: 代码随想录训练营/day1.py
def search(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: int
    """
    # 二分查找
    left, right = 0, len(nums) - 1
    while left <= right:
        mid = (left + right) // 2
        if target == nums[mid]:
            return mid
        elif target < nums[mid]:
            right = mid - 1
        else:
            left = mid + 1
    return -1

def removeElement(nums, target):
    slow = 0
    fast = 0
    while slow < fast:
        if nums[slow] != target:
            nums[slow] = nums[fast]
            slow += 1
        fast += 1
    return slow


n = int(input())
nums = [int(input()) for _ in range(n)]
target = int(input())
print(search(nums, target))